### PR TITLE
fix: [PL-23440]: Empty password in SMTP

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31356)

### CODE REVIEW

**Review by pbarapatre10 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31356)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Comment by VikasMaddukuriHarness** [comment]
trigger messagecheck

File: 400-rest/src/main/java/software/wings/resources/SettingResourceNg.java [hunk 1]
```diff
@@ -87,7 +87,7 @@
     SmtpConfig smtpConfig = (SmtpConfig) variable.getValue();
     SecretManagerConfig secretManagerConfig = secretManagerConfigService.getDefaultSecretManager(accountId);
     SecretText secretText = new SecretText();
-    String password = new String(smtpConfig.getPassword());
+    String password = (smtpConfig.getPassword() == null) ? "" : new String(smtpConfig.getPassword());
```
Comment (manish-manchanda) [inline 400-rest/src/main/java/software/wings/resources/SettingResourceNg.java:90]: Can unit test be written for the same?

**Review by manish-manchanda — APPROVED** [review]

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31356)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[400-rest/src/main/java/software/wings/resources/SettingResourceNg.java, line 90 at r1](https://reviewable.io/reviews/harness/harness-core/31356#-MxnaEWVm4a8kd5i2azu-r1-90:-Mxt1rExEf5JOAnbgXxO:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3d742670026dfee96f971887cd64f1330f93edb4/400-rest/src/main/java/software/wings/resources/SettingResourceNg.java#L90)):*
<details><summary><i>Previously, manish-manchanda wrote…</i></summary><blockquote>

Can unit test be written for the same?
</blockquote></details>

Done.

**Review by pbarapatre10 — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31356)* status: 2 file(s) to review. 1. unresolved discussions.

**Review by manish-manchanda — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31356)* status: :shipit: complete! ✓

**Pull request merged**